templates/etc/systemd/system/etcd.service.j2

```diff
@@ -12,6 +12,7 @@ Description=etcd
 Documentation=https://github.com/coreos
 
 [Service]
+Environment=ETCD_UNSUPPORTED_ARCH={{etcd_architecture}}
```

Comment:
It seems that this environment variable is only needed for `arm` arch. So please add an `if` clause accordingly.

Reply:
So I thought about that and looked at the etcd codebase and they have some logic that removes the variable if it is a supported platform and keeps it there is it isn't an officially supported platform. I figured with that being the case it would make sense to keep the environment variable as is to cover any other unsupported platforms, so it could be for more than just arm. Let me know what you think.

Thanks!

Reply:
I already suspected that :wink: It is a little bit confusing if such an environment variable is set even using a supported platform. If we don't add an `if` clause then we should at least have a comment right before the variable setting which makes clear why this exists. Maybe there is also a link to a documentation of the variable that could be added to the comment.
Another option would be an Ansible variable like `etcd_allow_unsupported_archs: "false"` which sets the variable accordingly in the systemd service file if needed. That would make it a little bit more "visible". Personally I believe that over 90% of the users run on `amd64`. While `arm` is quite popular of course it's still not that common. So it's more an exception of the rule.
Both options are somehow valid. Not quite sure ATM what's the best one.

Reply:
I kinda like that 2nd idea -- that makes it to where there is no question that the user knows that they are or are not using an unsupported platform (if they don't set the flag, it'll fail to start, they'll see why in the logs, etc.). I'll work on going down that route hopefully today. I'll also try to add some comments to reflect that appropriately.

Thanks!!
Shawn